pkg/operator/controller/certificate/controller.go

```diff
@@ -77,119 +59,56 @@ type reconciler struct {
 }
 
 func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	current, err := r.currentRouterCASecret()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if current != nil {
-		return reconcile.Result{}, nil
-	}
-	desired, err := desiredRouterCASecret(r.operatorNamespace)
+	ca, err := r.ensureRouterCASecret()
 	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if err := r.createRouterCASecret(desired); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to create CA secret: %v", err)
-	}
-	r.recorder.Event(desired, "Normal", "CreatedWildcardCACert", "Created a default wildcard CA certificate")
-	return reconcile.Result{}, nil
-}
-
-// currentRouterCASecret returns the current router CA secret.
-func (r *reconciler) currentRouterCASecret() (*corev1.Secret, error) {
-	name := CASecretName(r.operatorNamespace)
-	secret := &corev1.Secret{}
-	if err := r.client.Get(context.TODO(), name, secret); err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil
+		return reconcile.Result{}, fmt.Errorf("failed to ensure router CA secret: %v", err)
+	}
+
+	result := reconcile.Result{}
+	errs := []error{}
+	ingress := &ingressv1alpha1.ClusterIngress{}
+	if err := r.client.Get(context.TODO(), request.NamespacedName, ingress); err != nil {
+		if !errors.IsNotFound(err) {
```

Comment:
Looks like `!errors.IsNotFound(err)` should be `errors.IsNotFound(err)`.

Reply:
Good catch, fixed